src/libcore/intrinsics.rs

```diff
@@ -869,11 +874,12 @@ extern "rust-intrinsic" {
     /// // the original inner type (`&i32`) to the converted inner type
     /// // (`Option<&i32>`), so read the nomicon pages linked above.
     /// let v_from_raw = unsafe {
-    ///     Vec::from_raw_parts(v_orig.as_mut_ptr() as *mut Option<&i32>,
-    ///                         v_orig.len(),
-    ///                         v_orig.capacity())
+    ///     // Ensure the original vector is not dropped.
+    ///     let v_clone: &mut Vec<_> = &mut *std::mem::ManuallyDrop::new(v_clone);
```

Comment:
```suggestion
    ///     let v_clone = std::mem::ManuallyDrop::new(v_clone);
```
This should work due to the `Deref`, right? And I think it is more readable.